### PR TITLE
Unify code creating debug source documents across C# and VB

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera FYI

File: src/Compilers/CSharp/Portable/Errors/MessageProvider.cs [hunk 1]
```diff
@@ -177,36 +177,37 @@ public override void ReportDuplicateMetadataReferenceWeak(DiagnosticBag diagnost
...
 
         // pseudo-custom attributes:
-        public override int ERR_PermissionSetAttributeFileReadError { get { return (int)ErrorCode.ERR_PermissionSetAttributeFileReadError; } }
+        public override int ERR_PermissionSetAttributeFileReadError => (int)ErrorCode.ERR_PermissionSetAttributeFileReadError;
 
         // PDB Writer:
-        public override int WRN_PdbUsingNameTooLong { get { return (int)ErrorCode.WRN_DebugFullNameTooLong; } }
-        public override int WRN_PdbLocalNameTooLong { get { return (int)ErrorCode.WRN_PdbLocalNameTooLong; } }
-        public override int ERR_PdbWritingFailed { get { return (int)ErrorCode.FTL_DebugEmitFailure; } }
+        public override int ERR_EncodinglessSyntaxTree => (int)ErrorCode.ERR_EncodinglessSyntaxTree;
```
Comment (tmat) [inline src/Compilers/CSharp/Portable/Errors/MessageProvider.cs]: Added this line. The rest is just clean-up.

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/Core/Portable/Compilation/Compilation.cs [hunk 2]
```diff
@@ -1527,6 +1527,58 @@ internal bool IsRealSigned
             Predicate<ISymbol> filterOpt,
             CancellationToken cancellationToken);
 
+        internal bool StartSourceChecksumCalculation(DebugDocumentsBuilder documentsBuilder, DiagnosticBag diagnostics)
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs]: >StartSourceChecksumCalculation [](start = 22, length = 30)

CalculateSourceChecksum?  In general I expect Start methods to have End pairs.  Not the case here so we should choose a more representative name.

**Comment by jaredpar** [comment]
Didn't i essentially already review this as a part of the removal of `IAssembly`?

**Comment by tmat** [comment]
Yes, I think you did.

**Comment by jaredpar** [comment]
I think will once againc sign off :+1:

**Comment by agocke** [comment]
LGTM

**Pull request merged**